src/realm/sync/config.hpp

```diff
@@ -224,6 +224,9 @@ struct SyncConfig {
 
     bool simulate_integration_error = false;
 
+    std::function<void(std::shared_ptr<Realm>)> subscription_initializer{nullptr};
+    bool always_run = false;
```

Comment:
I am not sure about this flag also, it seems we may need to let SDKs decide if we want to run this every time or only in certain cases. The same behaviour can be obtained by setting the callback itself, though. 
I am not sure whether we should try to understand if it is the first time that we are opening the file and change this flag accordingly.